### PR TITLE
CORE-6782

### PR DESCRIPTION
**Pull request by MrMatthewLayton**
Added StatePointer Interfaces, Implementations and Tests

### CODE REVIEW

**Review by mnesbit — COMMENTED** [review]
I feel this is an advanced feature that can come in later. I am not sure if this couldn't in fact be deleted if we change way builder works

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/StatePointers.kt [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+/**
+ * Represents a state pointer that points statically to a specific [StateRef].
+ *
+ * @constructor Creates a new instance of the [StaticPointer] data class.
+ * @property identifier The identifier of the [ContractState] being pointed to.
+ * @property type The type of the [ContractState] being pointed to.
+ * @property isResolved Determines whether the current pointer should be resolved to a reference input when used in a transaction.
+ */
+data class StaticPointer<T : ContractState>(
```
Comment (lankydan) [inline ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/StatePointers.kt:28]: With the amount of code in these classes, they should be split into separate files with a appropriate package.

**Review by yarivus — COMMENTED** [review]
LGTM

**Comment by blsemo** [comment]
I agree with @mnesbit that we probably don't need that as a day 1 feature.

**Pull request closed without merging**